source/ShopPanel.h

```diff
@@ -74,36 +75,47 @@ class ShopPanel : public Panel {
 protected:
 	void DrawShip(const Ship &ship, const Point &center, bool isSelected);
 
-	void CheckForMissions(Mission::Location location);
+	static void DrawTooltip(const std::string &text, const Point &hoverPoint, const Color &textColor,
+		const Color &backColor);
+
+	void CheckForMissions(Mission::Location location) const;
 
 	// These are for the individual shop panels to override.
 	virtual int TileSize() const = 0;
-	virtual int VisibilityCheckboxesSize() const;
+	virtual int VisibilityCheckboxesSize() const { return 0; };
```

Comment:
No function bodies in the class body (you were probably misleaded by the BuyResult class, but that one's also wrong).